docs/migrating.rst

```diff
@@ -0,0 +1,275 @@
+######################################
+Migrating from Python 2 version of NDB
+######################################
+
+While every attempt has been made to keep compatibility with the previous
+version of `ndb`, there are fundamental differences at the platform level,
+which have made necessary in some cases to depart from the original
+implementation, and sometimes even to remove existing functionality
+altogether.
+
+One of the main objectives of this rewrite was to enable `ndb` for use in any
+Python environment, not just Google App Engine. As a result, many of the `ndb`
+APIs that relied on GAE environment and runtime variables, resources, and
+legacy APIs have been dropped.
+
+Aside from this, there are many differences between the Datastore APIs
+provided by GAE and those provided by the newer Google Cloud Platform. These
+differences have required some code and API changes as well.
+
+Finally, in many cases, new features of Python 3 have eliminated the need for
+some code, particularly from the old `utils` module.
+
+If you are migrating code, these changes can generate some confusion. This
+document will cover the most common migration issues.
+
+Setting up a connection
+=======================
+
+The most important difference from the previous `ndb` version, is that the new
+`ndb` requires the use of a client to set up a runtime context for a project.
+This is necessary because `ndb` can now be used in any Python environment, so
+we can no longer assume it's running in the context of a GAE request.
+
+The `ndb` client uses ``google.auth`` for authentication, which is how APIs in
+Google Cloud Platform work. The client can take a `credentials` parameter or
```

Comment:
Recommend "The `ndb` client uses ``google.auth`` for authentication, consistent with other Google Cloud Platform client libraries."